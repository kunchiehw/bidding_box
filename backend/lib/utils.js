const jwt = require('jsonwebtoken');

const secret = process.env.SHARE_SECRET;
const db = {
  users: {
    wkc: 'password',
    jarron: 'password',
  },
};


function authenticateUser(username, password) {
  return new Promise((resolve, reject) => {
    if (!username || !password || !(username in db.users) || db.users[username] !== password) {
      return reject();
    }

    const token = jwt.sign({ username }, secret, { expiresIn: '30d' });
    return resolve(token);
  });
}
module.exports.authenticateUser = authenticateUser;


function validateJwt(token) {
  return new Promise((resolve, reject) => {
    jwt.verify(token, secret, (err, decoded) => {
      if (err || !decoded) {
        reject();
      }

      if (!('username' in decoded) || !(decoded.username in db.users)) {
        reject();
      }

      resolve(token);
    });
  });
}
module.exports.validateJwt = validateJwt;


module.exports.validateJwtMiddleware = (req, res, next) => {
  if (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer') {
    const token = req.headers.authorization.split(' ')[1];
    validateJwt(token)
      .then(() => next())
      .catch(() => next(new Error('Unauthorized')));
  } else {
    next(new Error('Unauthorized'));
  }
};


module.exports.getTtl = () =>
  Math.floor(Date.now() / 1000) + (4 * 60 * 60); // ttl for 4 hour
