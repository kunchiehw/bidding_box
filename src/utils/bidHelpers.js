const { STANDARD_SUITS } = require('./constants');

function isPass(bid) {
  return (bid && bid.suit === 'PASS');
}

function isDouble(bid) {
  return (bid && bid.suit === 'DOUBLE');
}

function isSuit(bid) {
  return (bid && STANDARD_SUITS.includes(bid.suit));
}

function isBidSeqEnded(bidSeq) {
  const bidSeqLen = bidSeq.length;
  if (isPass(bidSeq[bidSeqLen - 3]) &&
      isPass(bidSeq[bidSeqLen - 2]) &&
      isPass(bidSeq[bidSeqLen - 1])) {
    return (bidSeqLen >= 4);
  }
  return false;
}

function isDoubleAllowed(bidSeq) {
  const bidSeqLen = bidSeq.length;
  if (isSuit(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  if (isSuit(bidSeq[bidSeqLen - 3]) &&
      isPass(bidSeq[bidSeqLen - 2]) &&
      isPass(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  return true;
}

function isRedoubleAllowed(bidSeq) {
  const bidSeqLen = bidSeq.length;
  if (isDouble(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  if (isDouble(bidSeq[bidSeqLen - 3]) &&
      isPass(bidSeq[bidSeqLen - 2]) &&
      isPass(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  return true;
}

function getCurrentBid(bidSeq) {
  for (let i = bidSeq.length - 1; i >= 0; i -= 1) {
    if (isSuit(bidSeq[i])) {
      return bidSeq[i];
    }
  }
  return { level: 0, suit: 'PASS' };
}

function isStandardBidAllowed(currentLevel, currentSuit, level, suit) {
  if (currentLevel !== level) {
    return (currentLevel > level);
  }
  return (STANDARD_SUITS.indexOf(currentSuit) >= STANDARD_SUITS.indexOf(suit));
}


module.exports = {
  isPass,
  isDouble,
  isSuit,
  isBidSeqEnded,
  isDoubleAllowed,
  isRedoubleAllowed,
  getCurrentBid,
  isStandardBidAllowed,
};
