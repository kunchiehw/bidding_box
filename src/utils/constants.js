export const POSSIBLE_LEVELS = [1, 2, 3, 4, 5, 6, 7];

export const STANDARD_SUITS = ['CLUBS', 'DIAMONDS', 'HEARTS', 'SPADES', 'NOTRUMPS'];
export const SPECIAL_SUITS = ['PASS', 'DOUBLE', 'REDOUBLE'];
export const ALL_SUITS = ['CLUBS', 'DIAMONDS', 'HEARTS', 'SPADES', 'NOTRUMPS', 'PASS', 'DOUBLE', 'REDOUBLE'];

export const NS_SEATS = ['NORTH', 'SOUTH'];
export const EW_SEATS = ['WEST', 'EAST'];
export const SEATS = ['WEST', 'NORTH', 'EAST', 'SOUTH'];
export const DECLARERS = ['WEST', 'NORTH', 'EAST', 'SOUTH', 'NS', 'EW'];
export const PARTICIPANTS = ['WEST', 'EAST', 'WATCHER'];

export const VULNERABILITIES = ['NONE', 'NS', 'EW', 'BOTH'];

export const BOOLEANS = [true, false];

export const SERVER_API_HOST = `${process.env.REACT_APP_BACKEND_SCHEMA}://${process.env.REACT_APP_BACKEND_URL}`;
export const SERVER_WS_HOST = `ws://${process.env.REACT_APP_BACKEND_URL}`;
