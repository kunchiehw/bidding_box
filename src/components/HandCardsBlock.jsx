import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import HandCardsCell from './HandCardsCell';
import { PARTICIPANTS } from '../utils/constants';

const propTypes = {
  playerRole: PropTypes.oneOf(PARTICIPANTS).isRequired,
  // TODO: Make more specific validator
  eastHand: PropTypes.shape({
    SPADES: PropTypes.string.isRequired,
    HEARTS: PropTypes.string.isRequired,
    DIAMONDS: PropTypes.string.isRequired,
    CLUBS: PropTypes.string.isRequired,
  }),
  westHand: PropTypes.shape({
    SPADES: PropTypes.string.isRequired,
    HEARTS: PropTypes.string.isRequired,
    DIAMONDS: PropTypes.string.isRequired,
    CLUBS: PropTypes.string.isRequired,
  }),
  eastId: PropTypes.string,
  westId: PropTypes.string,
  bidSeqIsEnded: PropTypes.bool.isRequired,
  handleClickToSit: PropTypes.func.isRequired,
};

const defaultProps = {
  eastHand: null,
  westHand: null,
  eastId: null,
  westId: null,
};

class HandCardsBlock extends PureComponent {
  render() {
    return (
      <Grid className="hand-cards-display-block" columns="equal">
        <Grid.Column>
          <HandCardsCell
            seat="WEST"
            playerRole={this.props.playerRole}
            playerId={this.props.westId}
            playerHand={this.props.westHand}
            bidSeqIsEnded={this.props.bidSeqIsEnded}
            handleClickToSit={this.props.handleClickToSit}
          />
        </Grid.Column>
        <Grid.Column>
          <HandCardsCell
            seat="EAST"
            playerRole={this.props.playerRole}
            playerId={this.props.eastId}
            playerHand={this.props.eastHand}
            bidSeqIsEnded={this.props.bidSeqIsEnded}
            handleClickToSit={this.props.handleClickToSit}
          />
        </Grid.Column>
      </Grid>
    );
  }
}

HandCardsBlock.propTypes = propTypes;
HandCardsBlock.defaultProps = defaultProps;

export default HandCardsBlock;
