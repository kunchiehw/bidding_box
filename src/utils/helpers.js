import { STANDARD_SUITS } from './constants';
import getMessage from '../i18n/messages';

export function bidToString(level, suit) {
  if (STANDARD_SUITS.includes(suit)) {
    return `${level} ${getMessage(suit)}`;
  }
  return getMessage(suit);
}

export function suitColor(suit) {
  switch (suit) {
    case 'NOTRUMPS':
      return 'black';
    case 'SPADES':
    case 'REDOUBLE':
      return 'blue';
    case 'HEARTS':
    case 'DOUBLE':
      return 'red';
    case 'DIAMONDS':
      return 'orange';
    case 'CLUBS':
    case 'PASS':
      return 'green';
    default:
      return '';
  }
}

export function shouldStandardBidButtonDisabled(currentLevel, currentSuit, buttonLevel, buttonSuit) {
  if (currentLevel !== buttonLevel) {
    return (currentLevel > buttonLevel);
  }
  return (STANDARD_SUITS.indexOf(currentSuit) >= STANDARD_SUITS.indexOf(buttonSuit));
}
