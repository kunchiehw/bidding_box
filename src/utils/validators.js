import { POSSIBLE_LEVELS, STANDARD_SUITS } from './constants';

// HOC for all validator
function createChainableTypeChecker(validate) {
  function checkType(isRequired, props, propName, componentName, location) {
    componentName = componentName || 'ANONYMOUS';
    if (props[propName] == null) {
      if (isRequired) {
        return new Error(`'${componentName}': requires prop '${propName}'.`);
      }
      return null;
    }
    return validate(props, propName, componentName, location);
  }

  const chainedCheckType = checkType.bind(null, false);
  chainedCheckType.isRequired = checkType.bind(null, true);

  return chainedCheckType;
}

export const levelValidator = createChainableTypeChecker((props, propName, componentName) => {
  componentName = componentName || 'ANONYMOUS';

  if (props.suit == null) {
    return new Error(`'${componentName}': requires prop 'suit' when validates prop '${propName}'.`);
  }

  if (props[propName]) {
    const level = props[propName];
    if (STANDARD_SUITS.includes(props.suit)) {
      if (!POSSIBLE_LEVELS.includes(level)) {
        return new Error(`'${componentName}': '${propName}' should be in POSSIBLE_LEVELS when 'suit' is in STANDARD_SUITS.`);
      }
    } else if (level !== 0) {
      return new Error(`'${componentName}': '${propName}' should be 0 when 'suit' is in SPECIAL_SUITS.`);
    }
  }

  return null;
});

export function foo() {}
