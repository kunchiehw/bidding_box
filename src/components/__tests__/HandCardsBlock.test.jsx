import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HandCardsBlock from '../HandCardsBlock';
import { PARTICIPANTS, BOOLEANS } from '../../utils/constants';

describe('HandCardsDisplay render', () => {
  const hands = [
    {
      eastHand: null,
      westHand: null,
    },
    {
      eastHand: {
        SPADES: 'AT4', HEARTS: 'K97', DIAMONDS: 'Q875', CLUBS: 'KJT',
      },
      westHand: {
        SPADES: 'QJ632', HEARTS: 'AT653', DIAMONDS: 'K9', CLUBS: '2',
      },
    },
  ];
  const ids = [
    {
      eastId: 'jarron',
      westId: 'wkc',
    },
    {
      eastId: 'lonely',
      westId: '',
    },
  ];

  PARTICIPANTS.forEach((playerRole) => {
    ids.forEach((playerIds) => {
      hands.forEach((playerHands) => {
        BOOLEANS.forEach((bidSeqIsEnded) => {
          it(`playerRole: ${playerRole}, playerIds: ${playerIds},
              playerHands: ${playerHands}, bidSeqIsEnded: ${bidSeqIsEnded}`, () => {
            const handCardsDisplay = shallow(<HandCardsBlock
              playerRole={playerRole}
              eastHand={playerHands.eastHand}
              westHand={playerHands.westHand}
              eastId={playerIds.eastId}
              westId={playerIds.westId}
              bidSeqIsEnded={bidSeqIsEnded}
              handleClickToSit={() => {}}
            />);
            expect(toJson(handCardsDisplay)).toMatchSnapshot();
          });
        });
      });
    });
  });
});
