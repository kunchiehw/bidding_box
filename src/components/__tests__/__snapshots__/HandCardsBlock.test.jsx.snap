// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 1`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="wkc"
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="jarron"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 2`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId="wkc"
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="jarron"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 3`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId=""
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="lonely"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 4`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId=""
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="lonely"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 1`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="wkc"
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="jarron"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 2`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId="wkc"
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="jarron"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 3`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId=""
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="lonely"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: EAST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 4`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId=""
      playerRole="EAST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="lonely"
      playerRole="EAST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 1`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="wkc"
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="jarron"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 2`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId="wkc"
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="jarron"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 3`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId=""
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="lonely"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 4`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId=""
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="lonely"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 1`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="wkc"
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="jarron"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 2`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId="wkc"
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="jarron"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 3`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId=""
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="lonely"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WATCHER, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 4`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId=""
      playerRole="WATCHER"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="lonely"
      playerRole="WATCHER"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 1`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="wkc"
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="jarron"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 2`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId="wkc"
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="jarron"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 3`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId=""
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="lonely"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: false 4`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId=""
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={false}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="lonely"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 1`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="wkc"
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="jarron"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 2`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId="wkc"
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="jarron"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 3`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId=""
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={null}
      playerId="lonely"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;

exports[`HandCardsDisplay render playerRole: WEST, playerIds: [object Object],
              playerHands: [object Object], bidSeqIsEnded: true 4`] = `
<Grid
  className="hand-cards-display-block"
  columns="equal"
>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "2",
          "DIAMONDS": "K9",
          "HEARTS": "AT653",
          "SPADES": "QJ632",
        }
      }
      playerId=""
      playerRole="WEST"
      seat="WEST"
    />
  </GridColumn>
  <GridColumn>
    <HandCardsCell
      bidSeqIsEnded={true}
      handleClickToSit={[Function]}
      playerHand={
        Object {
          "CLUBS": "KJT",
          "DIAMONDS": "Q875",
          "HEARTS": "K97",
          "SPADES": "AT4",
        }
      }
      playerId="lonely"
      playerRole="WEST"
      seat="EAST"
    />
  </GridColumn>
</Grid>
`;
